packages/api/src/hooks/p2p/useCreateAdvertiser.ts

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useMemo } from 'react';
+import useAuthorize from '../useAuthorize';
+import useMutation from '../../useMutation';
+import useInvalidateQuery from '../../useInvalidateQuery';
+
+type TAdvertiseCreatePayload = {
+    name: string;
```

Comment:
```suggestion
 name: NonNullable<
        Parameters<ReturnType<typeof useMutation<'p2p_advertiser_create'>>['mutate']>
    >[0]['payload']['name'];
```

Reply:
@thisyahlen-deriv I initially attempted that approach, but TypeScript interpreted it as `any` (same for other hooks that used you approach) . that's why I just typed it as `string`.
Is there a particular reason for preferring the typing approach you suggested?